project_forecast_line_deadline/tests/test_forecast_line_deadline.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Therp BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import odoo.tests.common as common
+
+
+class TestForecastLineDeadline(common.TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.project_task = self.env.ref("project.project_task_1")
+
+    def test_forecast_end_deadline(self):
+        """Test forecast_date_planned_end vs date_deadline"""
+        task = self.project_task
+        # See that relevant date fields are falsy
+        self.assertFalse(task.forecast_date_planned_end)
+        self.assertFalse(task.date_deadline)
+        # Set deadline for task
+        task.date_deadline = "2022-01-01"
+        # Date fields are equal
+        self.assertEqual(task.forecast_date_planned_end, task.date_deadline)
+        # Set forecast_date_deadline manually
+        task.forecast_date_planned_end = "2022-02-01"
+        # Dates are not the same anymore
+        self.assertNotEqual(task.forecast_date_planned_end, task.date_deadline)
```

Comment:
Hm, "not the same anymore" could also mean that None was written to the field. Could we, aside from this, also explicitly check for the date that was written?

Reply:
Basically we give a value directly on the field in line 23.

Reply:
...yes, but we then don't check for that same value. So, we don't test whether the inverse function actually writes that value - we just test whether the result is a different value. (It's nitpicking, I know)